consensus/istanbul/core/prepare_test.go

```diff
@@ -81,3 +84,38 @@ func TestCore_sendPrepare(t *testing.T) {
 		mockCtrl.Finish()
 	}
 }
+
+func BenchmarkMsgCmp(b *testing.B) {
+	getEmptySubject := func() istanbul.Subject {
+		return istanbul.Subject{
+			View: &istanbul.View{
+				Round:    big.NewInt(0),
+				Sequence: big.NewInt(0),
+			},
+			Digest:   common.HexToHash("1"),
+			PrevHash: common.HexToHash("2"),
+		}
+	}
+	s1, s2 := getEmptySubject(), getEmptySubject()
+
+	// Worst
+	b.Run("reflect.DeepEqual", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			reflect.DeepEqual(s1, s2)
+		}
+	})
+
+	// Better
+	b.Run("cmp", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			cmp.Equal(&s1, &s2)
+		}
+	})
+
+	// Best
+	b.Run("own", func(b *testing.B) {
```

Comment:
Could you make some test codes so this improvement does not make any problems with consensus? Thank you in advance.

Reply:
Also, I did not get the meaning of "own" here. Does it mean it is your own code?

Reply:
how about "precompiledEqual"?

Reply:
Hmm.., I think just a `EuqalImpl` seems enough. The suggested names seem overly named than it does actually.